cpp/src/barretenberg/proof_system/circuit_builder/circuit_simulator.hpp

```diff
@@ -0,0 +1,113 @@
+    static constexpr merkle::HashType merkle_hash_type = merkle::HashType::NONE; // UGH
+    static constexpr pedersen::CommitmentType commitment_type = pedersen::CommitmentType::NONE;
+
+    bool contains_recursive_proof = false;
+    static constexpr size_t UINT_LOG2_BASE = 2; // WORKTODO: 6 for Ultra
+
+    const uint32_t zero_idx = 0;
+
+    // uint32_t add_variable([[maybe_unused]]const FF& in){
+    //   return 0; // WORKTODO: return part of `in` for debugging purposes?
+    // }
+
+    inline uint32_t add_variable([[maybe_unused]] const barretenberg::fr index) const { return 1028; }
+    inline barretenberg::fr get_variable([[maybe_unused]] const uint32_t index) const { return 1028; }
+
+    uint32_t put_constant_variable([[maybe_unused]] const barretenberg::fr& variable) { return 1028; }
+    void set_public_input([[maybe_unused]] const uint32_t witness_index)
+    {
+        // WORKTODO Public input logic?
+    }
+
+    void fix_witness([[maybe_unused]] const uint32_t witness_index,
+                     [[maybe_unused]] const barretenberg::fr& witness_value){
+        // WORKTODO: logic?
+    };
+
+    [[nodiscard]] size_t get_num_gates() const { return 0; }
+
+    void create_add_gate([[maybe_unused]] const add_triple& in){};
+    void create_mul_gate([[maybe_unused]] const mul_triple& in){};
+    void create_bool_gate([[maybe_unused]] const uint32_t a){};
+    void create_poly_gate([[maybe_unused]] const poly_triple& in){};
+    void create_big_add_gate([[maybe_unused]] const add_quad& in){};
+    void create_big_add_gate_with_bit_extraction([[maybe_unused]] const add_quad& in){};
+    void create_big_mul_gate([[maybe_unused]] const mul_quad& in){};
+    void create_balanced_add_gate([[maybe_unused]] const add_quad& in){};
+    void create_fixed_group_add_gate([[maybe_unused]] const fixed_group_add_quad& in){};
+    void create_fixed_group_add_gate_with_init([[maybe_unused]] const fixed_group_add_quad& in,
+                                               [[maybe_unused]] const fixed_group_init_quad& init){};
+    void create_fixed_group_add_gate_final([[maybe_unused]] const add_quad& in){};
+
+    accumulator_triple create_and_constraint([[maybe_unused]] const uint32_t a,
+                                             [[maybe_unused]] const uint32_t b,
+                                             [[maybe_unused]] const size_t num_bits)
+    {
+        return { { 1028 }, { 1028 }, { 1028 } };
+    };
+    accumulator_triple create_xor_constraint([[maybe_unused]] const uint32_t a,
+                                             [[maybe_unused]] const uint32_t b,
+                                             [[maybe_unused]] const size_t num_bits)
+    {
+        return { { 1028 }, { 1028 }, { 1028 } };
+    };
+
+    size_t get_num_constant_gates() { return 1028; };
+    // maybe this shouldn't be implemented?
+
+    bool create_range_constraint(FF const& elt,
```

Comment:
My goal is to preserve the composer interface when I can, but without tracking witness indices. In this case I decided to change the interface (input is a field element `elt` rather than a witness index). One way to unify interfaces would be to pass around `witness_t`'s rather than indices, since this is a container for both an index and a vale. But this would undermine the optimization of tracking witness indices.